app/models/miq_expression.rb

```diff
@@ -575,7 +575,7 @@ def self._to_ruby(exp, context_type, tz)
   def to_sql(tz = nil)
     tz ||= "UTC"
     @pexp, attrs = preprocess_for_sql(@exp.deep_clone)
-    sql = to_arel(@pexp, tz).to_sql if @pexp.present?
+    sql = Expression.build(@pexp, tz).to_sql if @pexp.present?
```

Comment:
> Also, it's not clear why we need to pass the timezone in. From the looks of the classes below, some care about the tz, some don't.

Agreed...I'm wondering if I can find some way to inject the value in such a way that any possible timezone things have been pre-resolved. I think Arel is even smart enough to work out things with dates and ranges in a way that we might be trying to duplicate. More research needed!